samcli/hook_packages/terraform/hooks/prepare/hook.py

```diff
@@ -46,6 +47,7 @@
 PropertyBuilderMapping = Dict[str, PropertyBuilder]
 
 TERRAFORM_BUILD_SCRIPT = "copy_terraform_built_artifacts.py"
+TF_BACKEND_OVERRIDE_FILENAME = "z_samcli_backend_override.tf"
```

Comment:
as we discussed, let's name it `z_samcli_backend_override` and when we copy it in the makefile we can set the correct name